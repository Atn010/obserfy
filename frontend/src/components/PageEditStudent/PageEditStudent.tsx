import React, { FC, useEffect, useState } from "react"
import { navigate } from "gatsby"
import Box from "../Box/Box"
import { useQueryStudentDetails } from "../../hooks/students/useQueryStudentDetails"
import Input from "../Input/Input"
import Flex from "../Flex/Flex"
import Button from "../Button/Button"
import BackNavigation from "../BackNavigation/BackNavigation"
import LoadingPlaceholder from "../LoadingPlaceholder/LoadingPlaceholder"
import { getAnalytics } from "../../analytics"
import DeleteStudentDialog from "../DeleteStudentDialog/DeleteStudentDialog"
import Spacer from "../Spacer/Spacer"

interface Props {
  id: string
}
export const PageEditStudent: FC<Props> = ({ id }) => {
  const studentDetailUrl = `/dashboard/students/details?id=${id}`
  const [isDeletingStudent, setIsDeletingStudent] = useState(false)
  const [details] = useQueryStudentDetails(id)
  const [name, setName] = useState()

  useEffect(() => {
    setName(details?.name)
  }, [details])

  async function deleteStudent(): Promise<void> {
    const baseUrl = "/api/v1"

    const response = await fetch(`${baseUrl}/students/${id}`, {
      credentials: "same-origin",
      method: "DELETE",
    })

    if (response.status === 200) {
      navigate("/dashboard/home")
    }

    getAnalytics()?.track("Student Deleted", {
      responseStatus: response.status,
      studentName: name,
    })
  }

  async function updateStudent(): Promise<void> {
    const baseUrl = "/api/v1"
    const data = { name }

    const response = await fetch(`${baseUrl}/students/${id}`, {
      credentials: "same-origin",
      method: "PUT",
      body: JSON.stringify(data),
      headers: { "Content-Type": "application/json" },
    })

    if (response.status === 200) {
      navigate(studentDetailUrl)
    }

    getAnalytics()?.track("Student Updated", {
      responseStatus: response.status,
      studentName: name,
    })
  }

  const deleteStudentDialog = isDeletingStudent && (
    <DeleteStudentDialog
      student={{ id, name }}
      onConfirm={deleteStudent}
      onCancel={() => setIsDeletingStudent(false)}
    />
  )

  return (
    <>
      <Box maxWidth="maxWidth.sm" margin="auto">
        <BackNavigation to={studentDetailUrl} text="Details" />
        <Box mx={3} mt={3}>
          {details?.name ? (
            <Input
              label="Name"
              width="100%"
              value={name}
              onChange={e => setName(e.target.value)}
            />
          ) : (
            <Box pt={24}>
              <LoadingPlaceholder width="100%" height={56} />
            </Box>
          )}
        </Box>
        <Flex m={3}>
          <Spacer />
          <Button
            mr={2}
            variant="outline"
            color="danger"
            onClick={() => setIsDeletingStudent(true)}
          >
            Delete
          </Button>
          <Button onClick={updateStudent}>Save</Button>
        </Flex>
      </Box>
      {deleteStudentDialog}
    </>
  )
}

export default PageEditStudent
