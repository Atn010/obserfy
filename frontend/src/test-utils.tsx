/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { IntlProvider } from "gatsby-plugin-intl3"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"
import en from "./intl/en.json"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <IntlProvider locale="en" messages={en}>
          <Box fontSize={[16, 20]}>{children}</Box>
        </IntlProvider>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
