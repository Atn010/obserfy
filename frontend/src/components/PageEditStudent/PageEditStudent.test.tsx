import React from "react"
import { render } from "../../test-utils"
import { Basic } from "./PageEditStudent.stories"

describe("PageEditStudent", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
