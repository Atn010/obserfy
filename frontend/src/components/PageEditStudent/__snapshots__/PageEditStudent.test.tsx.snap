// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageEditStudent should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    background-color: rgba(165,165,165,0.05);
  }

  50% {
    background-color: rgba(165,165,165,0.1);
  }

  100% {
    background-color: rgba(165,165,165,0.05);
  }
}

.emotion-40 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width:40em) {
  .emotion-40 {
    font-size: 20px;
  }
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 8px;
  background-color: var(--theme-ui-colors-background,#F7F8FA);
  border-radius: 6px;
  box-shadow: rgba(0,0,0,0.1) 0px 0px 1px,rgba(0,0,0,0.27) 0px 2px 4px -2px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  z-index: 50;
  top: 0;
  position: fixed;
  border-radius: 0;
  padding: 0;
  background-color: var(--theme-ui-colors-surface,#fff);
  overflow: hidden;
  height: 57px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  height: 57px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 0%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:40em) {
  .emotion-2 {
    width: auto;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon,rgba(0,0,0,0.6));
  margin: 16px;
  height: 24px;
  width: 24px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  white-space: nowrap;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  font-size: 16px;
  margin-left: -4px;
  margin-bottom: 0;
}

@media screen and (min-width:40em) {
  .emotion-1 {
    font-size: 14px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  pointer-events: inherit;
  max-width: 56px;
}

@media screen and (min-width:40em) {
  .emotion-4 {
    pointer-events: none;
  }
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  -webkit-transition: margin-left 250ms cubic-bezier(0.0,0.0,0.2,1);
  transition: margin-left 250ms cubic-bezier(0.0,0.0,0.2,1);
  fill: var(--theme-ui-colors-icon,rgba(0,0,0,0.6));
  margin: 16px;
  height: 24px;
  width: 24px;
  margin-left: 16px;
  opacity: 1;
}

@media screen and (min-width:40em) {
  .emotion-3 {
    width: 0%;
    margin-left: 100px;
    opacity: 0;
  }
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 24px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 300;
  line-height: heading;
  font-size: 2rem;
  line-height: 3rem;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  font-weight: 400;
  opacity: 1;
  margin-bottom: 0;
  margin-left: 0;
}

@media screen and (min-width:40em) {
  .emotion-5 {
    opacity: 0;
    margin-left: 16px;
  }
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  pointer-events: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-style: solid;
  border-radius: 99999px;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  border-width: 1px;
  cursor: pointer;
  -webkit-transition: background-color 250ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background-color 250ms cubic-bezier(0.0,0.0,0.2,1);
  margin-left: 16px;
  margin-right: 16px;
  padding: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-10:hover {
  background-color: var(--theme-ui-colors-muted,#f6f6f9);
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  object-fit: cover;
  object-position: center;
  border-radius: 99999px;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  border-width: 1px;
  border-style: solid;
  overflow: hidden;
  width: 32px;
  height: 32px;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  font-size: 16px;
  margin-left: 8px;
  margin-right: 4px;
  margin-top: 0;
  margin-bottom: 0;
  display: none;
}

@media screen and (min-width:40em) {
  .emotion-8 {
    font-size: 14px;
    display: block;
  }
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon,rgba(0,0,0,0.6));
  margin: 0;
  height: 20px;
  width: 20px;
  margin-left: 8px;
  margin-right: 8px;
  opacity: 0.4;
}

@media screen and (min-width:40em) {
  .emotion-9 {
    margin-left: 0;
    margin-right: 4px;
  }
}

.emotion-25 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 360px;
  min-width: 230px;
  position: fixed;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  box-shadow: none;
  -webkit-transition: -webkit-transform 250ms cubic-bezier(0.0,0.0,0.2,1);
  -webkit-transition: transform 250ms cubic-bezier(0.0,0.0,0.2,1);
  transition: transform 250ms cubic-bezier(0.0,0.0,0.2,1);
  top: 0;
  left: 0;
  z-index: 100;
  border-right-style: solid;
  border-right-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  border-right-width: 1px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: var(--theme-ui-colors-background,#F7F8FA);
  width: 82%;
  height: calc(100% + 57px);
  padding-bottom: 65px;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:40em) {
  .emotion-25 {
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
    box-shadow: none;
    z-index: 0;
  }
}

@media screen and (min-width:40em) {
  .emotion-25 {
    width: 230px;
  }
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  box-shadow: rgba(0,0,0,0.1) 0px 0px 1px,rgba(0,0,0,0.27) 0px 2px 4px -2px;
  height: 57px;
  background-color: var(--theme-ui-colors-surface,#fff);
  margin-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:40em) {
  .emotion-14 {
    opacity: 0;
  }
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 8px;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 8px;
  background-color: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-17:hover {
  background-color: var(--theme-ui-colors-primaryLightest,rgba(161,130,250,0.05));
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon,rgba(0,0,0,0.6));
  margin: 16px;
  height: 20px;
  width: 20px;
  margin-left: 0;
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  font-size: 16px;
  font-weight: normal;
  margin: 0;
}

@media screen and (min-width:40em) {
  .emotion-16 {
    font-size: 14px;
  }
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 8px;
  margin-top: 0;
  margin-bottom: 16px;
  padding-left: 8px;
  background-color: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-21:hover {
  background-color: var(--theme-ui-colors-primaryLightest,rgba(161,130,250,0.05));
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-style: none;
  cursor: pointer;
  border-radius: 6px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 8px;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-24:hover {
  background-color: var(--theme-ui-colors-primaryLightest,rgba(161,130,250,0.05));
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  font-size: 14px;
  margin: 0;
}

.emotion-26 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  display: inherit;
  pointer-events: none;
  position: fixed;
  top: 0;
  left: 0;
  min-height: 100vh;
  z-index: 82;
  -webkit-transition: background 250ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background 250ms cubic-bezier(0.0,0.0,0.2,1);
  background-color: #00000000;
  width: 100%;
  height: 100%;
}

@media screen and (min-width:40em) {
  .emotion-26 {
    display: none;
  }
}

.emotion-28 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  top: 0;
  right: 0;
  z-index: 150;
  height: 57px;
  width: 18vw;
  display: none;
}

@media screen and (min-width:40em) {
  .emotion-28 {
    display: none;
  }
}

.emotion-27 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-onOverlay,#fff);
  width: 36px;
  height: 36px;
}

.emotion-39 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  margin-top: 57px;
  padding-left: 0;
  background-color: var(--theme-ui-colors-background,#F7F8FA);
}

@media screen and (min-width:40em) {
  .emotion-39 {
    padding-left: 230px;
  }
}

.emotion-38 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 640px;
  margin: auto;
}

.emotion-31 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-29 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon,rgba(0,0,0,0.6));
  margin: 16px;
  height: 24px;
  width: 24px;
  margin-right: 4px;
}

.emotion-30 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  font-size: 16px;
  margin-bottom: 0;
}

@media screen and (min-width:40em) {
  .emotion-30 {
    font-size: 14px;
  }
}

.emotion-34 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 16px;
  margin-right: 16px;
}

.emotion-33 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 24px;
}

.emotion-32 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-animation: 1s ease-in-out 0s infinite animation-0;
  animation: 1s ease-in-out 0s infinite animation-0;
  width: 100%;
  height: 56px;
}

.emotion-37 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-35 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary,#6E55C6);
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  text-transform: capitalize;
  box-sizing: border-box;
  font-size: 14px;
  font-weight: bold;
  color: var(--theme-ui-colors-textPrimary,#3A2B95);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  border: 1px solid;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 16px;
  color: var(--theme-ui-colors-danger,#d50000);
}

.emotion-35:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-35:hover,
.emotion-35:focus {
  background-color: var(--theme-ui-colors-primaryLighter,rgba(161,130,250,0.12));
  border-color: var(--theme-ui-colors-primary,#6E55C6);
  outline: none;
}

.emotion-35:hover:disabled,
.emotion-35:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
}

.emotion-36 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary,#6E55C6);
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  text-transform: capitalize;
  box-sizing: border-box;
  font-size: 14px;
  font-weight: bold;
  color: var(--theme-ui-colors-onPrimary,#fff);
  background-color: var(--theme-ui-colors-primary,#6E55C6);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-36:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-36:hover,
.emotion-36:focus {
  background-color: var(--theme-ui-colors-primaryDark,#3A2B95);
  outline: none;
}

.emotion-36:hover:disabled,
.emotion-36:focus:disabled {
  background-color: var(--theme-ui-colors-primary,#6E55C6);
}

<div>
  <div
    class="emotion-40"
  >
    <header
      class="emotion-11"
    >
      <div
        class="emotion-2"
      >
        <div
          alt="School Icon"
          class="emotion-0"
        />
        <div
          class="emotion-1"
        />
      </div>
      <div
        class="emotion-4"
      >
        <div
          alt="Menu"
          class="emotion-3"
          data-cy="menu"
        />
      </div>
      <h6
        class="emotion-5"
      >
        Home
      </h6>
      <div
        class="emotion-6"
      />
      <div
        class="emotion-10"
        data-cy="schoolName"
      >
        <div
          class="emotion-7"
        >
          <div
            class=" gatsby-image-wrapper"
            style="position: relative; overflow: hidden; display: inline-block; width: 32px; height: 32px;"
          >
            <img
              alt=""
              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsTAAALEwEAmpwYAAADvUlEQVQ4y7WU7VNUZRjG98/oSzON9ikTQUFidyHY1MDBlCgBqxmbUacGh3Eop9LsxSY0NN6FaUEQWCxCZYBKBxVLS1wQAiZCQNBVCY2MJHZh2XOeX/fZl0Bt+lRn5tpznjnn/s113c/9rIn/+DL9r0CllF/zz7pI898JrXUNtAXSA++CRfPA+0BG0QOX+jdbBtRQyOGDrkII3+Qg3juX0WcmYO4e2swg+q12ON+G+u40qusS3Bm/D2wKFS+MO3vjG+62pvJ78wqmWyPwOZNQPcnwYxxk21BrY2G9FTbIOiMJct+DuxNBhwbIQAZjugermfhyMZ6zEdAhhV1S5DRDbwwtb4RzJV6e09eg0gLixVWQGAVZm8HjDkUOwHTvNPda49AuPgXdCajOWHSnQPvjOFMQRYZ1OcUvPwuvPodKfQa1cTUY2rRWoCvhcLHhUPu74cpVinJGw+WnUR3WAKzXwrelYXz97lZ8J/LwtBThfUvcbIgXh0GgoecTYFtasIeaBzW8Gy4tB3FlwDQD1mPml+ZV1L/5Er7GQjoP7GDAvoee11IhJcHvUBmR08X1OgsU7xegx4Xq2wTt4QIznAmoQ/rWa0Y/n4a74SB/1h+U+wGGy3bx/afZDGxLgVTbfOSNwV7ecgmwO1GcRUpMieCMDQD7Y2nKXUrmmmiuHNrth/3xRS6TR/dxsWAn7uxX4AXpYYbUpotWS7KifYFdVk6rNF/siyu9R0ah28pErpWUJYswP/4IF3IyobmQqbqP+c2Rg6t4pzizQVI0KjkGZUQ1xmZ2Jjg2nYY7I24keqNEzUridqKNjGWPkhS+mLHDe+kqeIeJJjvTTSXMtR1HNTjQK0rQa+wyn53zJ8kYbO+1U7h/bmC2Tz5sr4eb5zhW8D5xix4jf6sNz4m9jDp2MTU2JJvnffjU+WdYGT9+qOm49Pdz6WmDHIRTW65ztaKMuS4bA2VZDH3UyrW8o4zub6Rt8whfbYGmzF859rqLszm3Ge9zP3TKTEdWaFRH+aiOVFRKcvsT8MP2kzAQwYWsFuxPQqV0pVL6XrVMURHmIz/8JrlhQ+RHjeCsmgw41TQ/1FRrVtTGIFI4LBoOq4/PBHq9XGKeTOOInCqHWfer1qKokz2otngpiRulyDLCJ0uG+allKvTvJUA/LACskRPnsPioWCqbvcfBzLlk6qxz1EQH3wdVK98dMo9RFHuVvJUuqtf3Mtl5Bs09tRAYcumjXIA9H1Qx07bOv/4nYKl5nELLMIVmF2XxffR/uJ0b5W/zF4A0F9hGaJU9AAAAAElFTkSuQmCC"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
            />
            <picture>
              <source
                srcset="/static/d765df87b7164d5d26f62ef45d04c06f/1bf60/avatar.webp 1x,
/static/d765df87b7164d5d26f62ef45d04c06f/46586/avatar.webp 1.5x,
/static/d765df87b7164d5d26f62ef45d04c06f/fdeef/avatar.webp 2x"
                type="image/webp"
              />
              <source
                srcset="/static/d765df87b7164d5d26f62ef45d04c06f/e381a/avatar.png 1x,
/static/d765df87b7164d5d26f62ef45d04c06f/1eb56/avatar.png 1.5x,
/static/d765df87b7164d5d26f62ef45d04c06f/cbd14/avatar.png 2x"
              />
              <img
                alt="avatar"
                height="32"
                loading="lazy"
                src="/static/d765df87b7164d5d26f62ef45d04c06f/e381a/avatar.png"
                srcset="/static/d765df87b7164d5d26f62ef45d04c06f/e381a/avatar.png 1x,
/static/d765df87b7164d5d26f62ef45d04c06f/1eb56/avatar.png 1.5x,
/static/d765df87b7164d5d26f62ef45d04c06f/cbd14/avatar.png 2x"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
                width="32"
              />
            </picture>
            <noscript>
              &lt;picture&gt;&lt;source type='image/webp' srcset="/static/d765df87b7164d5d26f62ef45d04c06f/1bf60/avatar.webp 1x,
/static/d765df87b7164d5d26f62ef45d04c06f/46586/avatar.webp 1.5x,
/static/d765df87b7164d5d26f62ef45d04c06f/fdeef/avatar.webp 2x" /&gt;&lt;source srcset="/static/d765df87b7164d5d26f62ef45d04c06f/e381a/avatar.png 1x,
/static/d765df87b7164d5d26f62ef45d04c06f/1eb56/avatar.png 1.5x,
/static/d765df87b7164d5d26f62ef45d04c06f/cbd14/avatar.png 2x" /&gt;&lt;img loading="lazy" width="32" height="32" srcset="/static/d765df87b7164d5d26f62ef45d04c06f/e381a/avatar.png 1x,
/static/d765df87b7164d5d26f62ef45d04c06f/1eb56/avatar.png 1.5x,
/static/d765df87b7164d5d26f62ef45d04c06f/cbd14/avatar.png 2x" src="/static/d765df87b7164d5d26f62ef45d04c06f/e381a/avatar.png" alt="avatar" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
            </noscript>
          </div>
        </div>
        <div
          class="emotion-8"
        />
        <div
          alt="Store Icon"
          class="emotion-9"
        />
      </div>
    </header>
    <nav
      class="emotion-25"
      data-cy="sidebar"
    >
      <div
        class="emotion-14"
      >
        <div
          alt="School Icon"
          class="emotion-0"
        />
        <div
          class="emotion-1"
        />
      </div>
      <a
        href="/"
      >
        <div
          class="emotion-17"
        >
          <div
            alt="Home icon"
            class="emotion-15"
          />
          <p
            class="emotion-16"
          >
            Home
          </p>
        </div>
      </a>
      <div
        class="emotion-6"
      />
      <a
        href="/settings"
      >
        <div
          class="emotion-21"
        >
          <div
            alt="settings icon"
            class="emotion-15"
          />
          <p
            class="emotion-16"
          >
            settings
          </p>
        </div>
      </a>
      <button
        class="emotion-24"
        data-cy="toggleTheme"
      >
        <div
          alt="Dark Mode icon"
          class="emotion-15"
        />
        <p
          class="emotion-23"
        >
          Dark Mode
        </p>
      </button>
    </nav>
    <div
      class="emotion-26"
      data-cy="outsideSidebar"
    />
    <div
      class="emotion-28"
    >
      <div
        class="emotion-27"
      />
    </div>
    <main
      class="emotion-39"
    >
      <div
        class="emotion-38"
      >
        <a
          href="/students/details?id=undefined"
        >
          <div
            class="emotion-31"
          >
            <div
              class="emotion-29"
            />
            <p
              class="emotion-30"
            >
              Details
            </p>
          </div>
        </a>
        <div
          class="emotion-34"
        >
          <div
            class="emotion-33"
          >
            <div
              class="emotion-32"
            />
          </div>
        </div>
        <div
          class="emotion-37"
        >
          <button
            class="emotion-35"
          >
            Delete
          </button>
          <button
            class="emotion-36"
          >
            Save
          </button>
        </div>
      </div>
    </main>
  </div>
</div>
`;
